appservice/src/SiteWrapper.ts

```diff
@@ -291,6 +293,36 @@ export class SiteWrapper {
         await requestP.get(`https://${this.defaultHostName}/admin/host/status`, requestOptions);
     }
 
+    public async getFile(client: KuduClient, filePath: string): Promise<IFileResult> {
+        // tslint:disable:no-unsafe-any
+        // tslint:disable-next-line:no-any
+        const response: any = (<any>await client.vfs.getItemWithHttpOperationResponse(filePath)).response;
+        if (response && response.body && response.headers && response.headers.etag) {
+            return { data: response.body, etag: response.headers.etag };
+            // tslint:enable:no-unsafe-any
+        } else {
+            throw new Error(localize('failedToFindFile', 'Failed to find file with path "{0}".', filePath));
+        }
+    }
+
+    /**
+     * Returns the latest etag of the updated file
+     */
+    public async putFile(client: KuduClient, data: Readable | string, filePath: string, etag: string): Promise<string> {
+        let stream: Readable;
+        if (typeof data === 'string') {
+            stream = new Readable();
+            stream._read = function (_size: number): void {
+                this.push(data);
+                this.push(null);
```

Comment:
What's the purpose of pushing a null at the end of the data stream?  Does it just cut the stream?

Reply:
Per [the docs](https://nodejs.org/api/stream.html#stream_readable_push_chunk_encoding)

"Passing chunk as null signals the end of the stream (EOF), after which no more data can be written."